src/ImageSharp/Common/Helpers/SimdUtils.cs

```diff
@@ -44,13 +48,24 @@ internal static Vector4 PseudoRound(this Vector4 v)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static Vector<float> FastRound(this Vector<float> v)
         {
-            var magic0 = new Vector<int>(int.MinValue); // 0x80000000
-            Vector<float> sgn0 = Vector.AsVectorSingle(magic0);
-            Vector<float> and0 = Vector.BitwiseAnd(sgn0, v);
-            Vector<float> or0 = Vector.BitwiseOr(and0, new Vector<float>(8388608.0f));
-            Vector<float> add0 = Vector.Add(v, or0);
-            Vector<float> sub0 = Vector.Subtract(add0, or0);
-            return sub0;
+#if SUPPORTS_RUNTIME_INTRINSICS
+
+            if (Avx.IsSupported)
+            {
+                ref Vector256<float> v256 = ref Unsafe.As<Vector<float>, Vector256<float>>(ref v);
+                Vector256<float> vRound = Avx.RoundToNearestInteger(v256);
+                return Unsafe.As<Vector256<float>, Vector<float>>(ref vRound);
+            }
```

Comment:
```suggestion
            if (Avx2.IsSupported)
            {
                ref Vector256<float> v256 = ref Unsafe.As<Vector<float>, Vector256<float>>(ref v);
                Vector256<float> vRound = Avx.RoundToNearestInteger(v256);
                return Unsafe.As<Vector256<float>, Vector<float>>(ref vRound);
            }
```

There is a call into this method from `Vector4Pair.RoundAndDownscalePreVector8` feeding 4-element 128 bit vectors the method, therefore We need to make sure that on old CPU-s having `Vector<float>.Count == 4`, this path is never executed.

If I remember it correctly AVX2 is needed for `Vector<float>.Count == 8`, so with that check we would be safe. @saucecontrol can you confirm?

Reply:
> AVX2 is needed for Vector<float>.Count == 8

Yup :)

Reply:
I was actually wondering about that. Thanks!